### PR TITLE
Added priority as an option

### PR DESCRIPTION
**Pull request by fahasch**

### CODE REVIEW

File: Sources/RemindersLibrary/CLI.swift [hunk 1]
```diff
@@ -42,12 +42,18 @@ private struct Add: ParsableCommand {
         name: .shortAndLong,
         help: "The date the reminder is due")
     var dueDate: DateComponents?
+    
+    @Option(
+        name: .shortAndLong,
+        help: "The priority of the reminder  [0: None, 4: High, 5: Medium, 6: Low]")
```
Comment (xingheng) [inline Sources/RemindersLibrary/CLI.swift:48]: Where are these raw values from?

File: Sources/RemindersLibrary/CLI.swift [hunk 2]
```diff
@@ -42,12 +42,18 @@ private struct Add: ParsableCommand {
...
+    var priority: Int
```
Comment (xingheng) [inline Sources/RemindersLibrary/CLI.swift:49]: Recommend defining a enum for this.

File: Sources/RemindersLibrary/Reminders.swift [hunk 3]
```diff
@@ -74,12 +74,13 @@ public final class Reminders {
         semaphore.wait()
     }
 
-    func addReminder(string: String, toListNamed name: String, dueDate: DateComponents?) {
+    func addReminder(string: String, toListNamed name: String, dueDate: DateComponents?, priority: Int) {
```
Comment (xingheng) [inline Sources/RemindersLibrary/Reminders.swift:77]: Giving it a default priority value will be great, I think.

**Review by xingheng — CHANGES_REQUESTED** [review]

File: Sources/RemindersLibrary/CLI.swift (hunk 1 above)
Comment (fahasch) [reply Sources/RemindersLibrary/CLI.swift]: These values are obtained by trial and error. I cannot find any documentation. [Property of EKReminder](https://developer.apple.com/documentation/eventkit/ekreminder) is simply an Int.

File: Sources/RemindersLibrary/CLI.swift (hunk 2 above)
Comment (fahasch) [reply Sources/RemindersLibrary/CLI.swift]: Thanks for the comment. I have updated the code.

File: Sources/RemindersLibrary/Reminders.swift [hunk 4]
```diff
@@ -74,12 +74,17 @@ public final class Reminders {
...
     }
 
-    func addReminder(string: String, toListNamed name: String, dueDate: DateComponents?) {
+    func addReminder(string: String, toListNamed name: String, dueDate: DateComponents?, priority: Priority?) {
         let calendar = self.calendar(withName: name)
         let reminder = EKReminder(eventStore: Store)
         reminder.calendar = calendar
         reminder.title = string
         reminder.dueDateComponents = dueDate
+        switch priority {
```
Comment (xingheng) [inline Sources/RemindersLibrary/Reminders.swift:83]: Suggest moving this switch expression to `Priority` as a computed property `value` or something others.

Comment (xingheng) [inline Sources/RemindersLibrary/Reminders.swift:83]: Suggest moving this switch expression to `Priority` as a computed property `value` or something others.

Comment (fahasch) [reply Sources/RemindersLibrary/Reminders.swift]: Thanks for the feedback. As these are my first attempts to write SWIFT, the code must look clumsy. I appreciate the input: learning by doing...

**Comment by keith** [comment]
Sorry I missed this one, I rebased and reworked it a bit here https://github.com/keith/reminders-cli/pull/42

**Pull request closed without merging**